app/src/nvm_buffer.rs

```diff
@@ -0,0 +1,137 @@
+use ledger_secure_sdk_sys::nvm_write;
+use utils::buffer::Writable;
+
+#[repr(align(64))]
+pub struct NVM<const N: usize>(pub [u8; N]);
+
+impl<const N: usize> NVM<N> {
+    pub const fn zeroed() -> Self {
+        Self([0; N])
+    }
+
+    pub fn write(&mut self, from: usize, slice: &[u8]) -> bool {
+        let len = slice.len();
+        if from + len > N {
+            return false;
+        }
+
+        unsafe {
+            let dst = self.0[from..].as_mut_ptr() as *mut _;
+            let src = slice.as_ptr() as *mut u8 as *mut _;
+            nvm_write(dst, src, len as u32); // TODO: handle error properly
+
+            debug_assert_eq!(&self.0[from..], &slice[..]);
```

Comment:
Great catch, I replaced it with `assert_eq!` since the ledger only supports building in release mode: https://github.com/LedgerHQ/ledger-device-rust-sdk/blob/3c22c1c1b5e2d909e34409fc92cfeed775541a63/cargo-ledger/src/main.rs#L226